bson/_cbsonmodule.c

```diff
@@ -2797,6 +2816,85 @@ static PyObject* _cbson_decode_all(PyObject* self, PyObject* args) {
     return result;
 }
 
+
+static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* args) {
+    uint32_t size;
+    uint32_t value_length;
+    uint32_t position = 0;
+    buffer_t buffer;
+    const char* string;
+    PyObject* arr;
+    PyObject* result = NULL;
+    Py_buffer view = {0};
+
+    if (!PyArg_ParseTuple(args, "O", &arr)) {
+        return NULL;
+    }
+
+    buffer = pymongo_buffer_new();
```

Comment:
We should use pymongo_buffer_save_space(size) to preallocate the max amount of space we need.

Reply:
Done